File: 001/BinaryGap.java
class Solution {
    public int solution(int N) {
        // 80%
        String binary = Integer.toBinaryString(N);
        int max = 0;
        int current = 0;
       for(int i = 0; i< binary.length(); i++) {
            if (binary.charAt(i) =='0') {
                current++;
            } else if (current > max) {
                max = current;
                current = 0;
            }
        }

        return max;
    }
}

File: 001/BinaryGap.rb

class BinaryGap

  def solution (number)

    binary = number.to_s(2)
    max = 0
    current = 0

    binary.each_char do |c|
      if c == '0'
        current += 1
      else
        max = current
        current = 0
      end
    end

    max
  end

end


#bg = BinaryGap.new
#puts "#{45633.to_s(2)} #{bg.solution(45633)}"
